lib/std/net.zig

```diff
@@ -450,24 +451,42 @@ pub fn getAddressList(allocator: *mem.Allocator, name: []const u8, port: u16) !*
             .next = null,
         };
         var res: *os.addrinfo = undefined;
-        switch (os.system.getaddrinfo(name_c.ptr, @ptrCast([*:0]const u8, port_c.ptr), &hints, &res)) {
-            @intToEnum(os.system.EAI, 0) => {},
-            .ADDRFAMILY => return error.HostLacksNetworkAddresses,
-            .AGAIN => return error.TemporaryNameServerFailure,
-            .BADFLAGS => unreachable, // Invalid hints
-            .FAIL => return error.NameServerFailure,
-            .FAMILY => return error.AddressFamilyNotSupported,
-            .MEMORY => return error.OutOfMemory,
-            .NODATA => return error.HostLacksNetworkAddresses,
-            .NONAME => return error.UnknownHostName,
-            .SERVICE => return error.ServiceUnavailable,
-            .SOCKTYPE => unreachable, // Invalid socket type requested in hints
-            .SYSTEM => switch (os.errno(-1)) {
-                else => |e| return os.unexpectedErrno(e),
-            },
-            else => unreachable,
+        const rc = sys.getaddrinfo(name_c.ptr, @ptrCast([*:0]const u8, port_c.ptr), &hints, &res);
+        if (builtin.os.tag == .windows) {
+            const ws2_32 = os.windows.ws2_32;
+            switch (rc) {
+                0 => {},
+                @enumToInt(ws2_32.WinsockError.WSATRY_AGAIN) => return error.TemporaryNameServerFailure,
+                @enumToInt(ws2_32.WinsockError.WSANO_RECOVERY) => return error.NameServerFailure,
+                @enumToInt(ws2_32.WinsockError.WSAEAFNOSUPPORT) => return error.AddressFamilyNotSupported,
+                @enumToInt(ws2_32.WinsockError.WSA_NOT_ENOUGH_MEMORY) => return error.OutOfMemory,
+                @enumToInt(ws2_32.WinsockError.WSAHOST_NOT_FOUND) => return error.UnknownHostName,
+                @enumToInt(ws2_32.WinsockError.WSATYPE_NOT_FOUND) => return error.ServiceUnavailable,
+                @enumToInt(ws2_32.WinsockError.WSAEINVAL) => unreachable,
+                @enumToInt(ws2_32.WinsockError.WSAESOCKTNOSUPPORT) => unreachable,
```

Comment:
Is there any reason in particular not to use `WSAGetLastError` here?  
I guess it isn't needed but it cleans up the code quite a bit.

Reply:
What do you mean by cleaner?

Reply:
It removes all the `@enumToInt` and `ws2_32.WinsockError.`, it's just a little nicer to read.  
Anyway, minor point and doing it this way is fine as well.

Reply:
How's this?

Reply:
👍